### ISSUE TITLE
Incorrectly reporting changes to the classes / interfaces

### ISSUE THREAD

**Issue by 243826** [siom79/japicmp#55]
I am evaluating the code on the master branch whether I can use this to follow the semantic versions for some artifacts we build. But I found out that there are false positives.

Here are the links to get the 2 jars:
http://www.malhar-inc.com/chetan/dt-api-2.0.0.jar
http://www.malhar-inc.com/chetan/dt-api-2.0.1.jar

And here is the command that I use (note that even if i change the association of new and old - in both cases it reports that methods are removed):
```
chetan@chubi:~/work/japicmp$ java -jar ./japicmp/target/japicmp-0.4.2-SNAPSHOT-jar-with-dependencies.jar -n /home/chetan/datatorrent/releases/2.0.1/lib/dt-api-2.0.1.jar -o /home/chetan/datatorrent/releases/2.0.0/lib/dt-api-2.0.0.jar -b
Comparing /home/chetan/datatorrent/releases/2.0.0/lib/dt-api-2.0.0.jar with /home/chetan/datatorrent/releases/2.0.1/lib/dt-api-2.0.1.jar:
***! MODIFIED CLASS: PUBLIC ABSTRACT com.datatorrent.api.LocalMode  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) com.datatorrent.api.DAG prepareDAG(com.datatorrent.api.StreamingApplication, org.apache.hadoop.conf.Configuration)
	---! REMOVED METHOD: PUBLIC(-) STATIC(-) void runApp(com.datatorrent.api.StreamingApplication, org.apache.hadoop.conf.Configuration, int)
***! MODIFIED INTERFACE: PUBLIC ABSTRACT com.datatorrent.api.StreamCodec  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) java.lang.Object fromByteArray(com.datatorrent.common.util.Slice)
***! MODIFIED INTERFACE: PUBLIC ABSTRACT com.datatorrent.api.StreamingApplication  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) void populateDAG(com.datatorrent.api.DAG, org.apache.hadoop.conf.Configuration)

chetan@chubi:~/work/japicmp$ java -jar ./japicmp/target/japicmp-0.4.2-SNAPSHOT-jar-with-dependencies.jar -o /home/chetan/datatorrent/releases/2.0.1/lib/dt-api-2.0.1.jar -n /home/chetan/datatorrent/releases/2.0.0/lib/dt-api-2.0.0.jar -b
Comparing /home/chetan/datatorrent/releases/2.0.1/lib/dt-api-2.0.1.jar with /home/chetan/datatorrent/releases/2.0.0/lib/dt-api-2.0.0.jar:
***! MODIFIED CLASS: PUBLIC ABSTRACT com.datatorrent.api.LocalMode  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) com.datatorrent.api.DAG prepareDAG(com.datatorrent.api.StreamingApplication, org.apache.hadoop.conf.Configuration)
	---! REMOVED METHOD: PUBLIC(-) STATIC(-) void runApp(com.datatorrent.api.StreamingApplication, org.apache.hadoop.conf.Configuration, int)
***! MODIFIED INTERFACE: PUBLIC ABSTRACT com.datatorrent.api.StreamCodec  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) java.lang.Object fromByteArray(com.datatorrent.common.util.Slice)
***! MODIFIED INTERFACE: PUBLIC ABSTRACT com.datatorrent.api.StreamingApplication  (not serializable)
	---! REMOVED METHOD: PUBLIC(-) ABSTRACT(-) void populateDAG(com.datatorrent.api.DAG, org.apache.hadoop.conf.Configuration)
```

**Comment by siom79** [comment]
Thank you for this detailed report. With the help of it I was able to reproduce the problem. It happens if you have two or more methods with the same name but different signatures in the same class whose parameters stem from a another jar file that is not on the classpath. Here the classes `com.datatorrent.api.StreamingApplication` and `org.apache.hadoop.conf.Configuration` are for example not on the classpath. In this case both classes have been as different ones.

As a workaround you can add these classes to the classpath.

Meanwhile I have also reworked the underlying functionality such that parameter classses are no longer resolved from the classpath. If you like you can checkout the `develop` branch and try it out:

```
git clone https://github.com/siom79/japicmp.git
cd japicmp/
git checkout develop
mvn clean install
```

Let me know if it works.

**Comment by 243826** [comment]
Hi Martin,

  Thanks for promptly looking at the code. Looks like the code in the
develop branch does not fix the issue. I was looking at it last night
myself and looks like your reason is right. But somehow your fix does not
address the issue.

  Let me know if you would like me to do anything to help fix this issue.

Thanks again.

--
Chetan

**Comment by siom79** [comment]
How do you invoke the tool?

In my case it prints that there are no changes:
```
>java -jar japicmp-0.4.2-SNAPSHOT-jar-with-dependencies.jar -o dt-api-2.0.0.jar -n dt-api-2.0.1.jar -m

Comparing dt-api-2.0.0.jar with dt-api-2.0.1.jar:
No changes.
```

**Comment by 243826** [comment]
I used -b flag not -m flag.

**Comment by 243826** [comment]
Sorry... did you add code since yesterday to fix the issue. I had not
updated my develop branch. I just did and will let you know if I still see
a problem.

--
Chetan

**Comment by 243826** [comment]
Ok... it worked. Thanks. I'll test more and let you know if I see any more
anomaly. Thanks for the prompt fix.

--
Regards,
Chetan

**Comment by siom79** [comment]
Released with version [0.5.0](https://github.com/siom79/japicmp/releases/tag/japicmp-base-0.5.0).
